### PR TITLE
add use-gpu option

### PR DESCRIPTION
**Pull request by viannecc**
Support "use-gpu" option when computes rnnlm perplexity.

### CODE REVIEW

File: scripts/rnnlm/compute_perplexity.sh [hunk 1]
```diff
@@ -39,7 +43,9 @@ done
 
 special_symbol_opts=$(cat $dir/special_symbol_opts.txt)
 
-ppl=$(rnnlm-sentence-probs --normalize-probs=true \
+opt="--normalize-probs=true --use-gpu=${use_gpu}"
+
+ppl=$(rnnlm-sentence-probs ${opt} \
        $special_symbol_opts $dir/final.raw "$word_embedding" \
        <(cat $text_in | sym2int.pl $dir/config/words.txt | awk '{print "utt_id ", $0}') | \
        awk '{for(i=2;i<=NF;i++) a+=$i; b+=NF-1}END{print exp(-a / b)}')
```
Comment (kkm000) [inline scripts/rnnlm/compute_perplexity.sh:51]: @viannecc, thanks, being able to pass the gpu option to the binary certainly won't hurt!

While reviewing the changes, I've spotted something that looked weird to me, a couple lines below. First `awk` adds a new field $1 to every line which is trivially 'utt_id', and the next `awk` in the pipeline sums up and counts fields _except_ the first, just added. I'd change this as below, but I want a second pair of eyes on the change. Can you think of anything wrong with it? I do not really know what this script really computes.
```suggestion
       <(cat $text_in | sym2int.pl $dir/config/words.txt |
         awk '{for(i=1;i<=NF;i++) a+=$i; b+=NF} END{print exp(-a / b)}' )
```
The `|` implies line continuation, `\` is not required, but that's purely cosmetic. It was the use of `awk` that baffled me.

File: scripts/rnnlm/compute_perplexity.sh [hunk 2]
```diff
@@ -3,10 +3,14 @@
...
 
 [ -f ./path.sh ] && . ./path.sh
+
+use_gpu=no
 . utils/parse_options.sh
 
 if [ $# != 2 ]; then
   echo "Usage: $0 <rnn-dir> <input-text>"
+  echo "Options: "
+  echo "  --use_gpu (yes|no|optional|wait)  # whether to use gpu."
```
Comment (kkm000) [inline scripts/rnnlm/compute_perplexity.sh:13]:
```suggestion
  echo "Usage: $0 [options] <rnn-dir> <input-text>"
  echo "Options: "
  echo "  --use_gpu (yes|no|optional|wait)  # whether to use gpu [no]."
```

File: scripts/rnnlm/compute_perplexity.sh (hunk 1 above)
Comment (viannecc) [reply scripts/rnnlm/compute_perplexity.sh:51]: Thanks for your fast review.

Comment (viannecc) [reply scripts/rnnlm/compute_perplexity.sh:51]: Hi. Thanks for your fast review！
I just checked the source code of rnnlm-sentence-probs.cc, which demands each line of the text to be evaluated should be started with a key, here is 'utt_id'. If without the key, the results would be wrong.  So, 'utt_id' is needed and 'i' should begin at 2. 
The 'NF' item denotes log prob of '</s>' symbol. So ’NF-1‘ is right I think.
'|' is pipeline symbol, it shows the results of 'rnnlm-sentence-probs' would be inputs of the last awk command. 
'\' implies line continuation.

Comment (kkm000) [reply scripts/rnnlm/compute_perplexity.sh:51]: Thanks for looking! I am not sure what is really going on, but as long as it works, let's not fix it! :)

**Comment by kkm000** [comment]
LGTM, thanks!

**Pull request merged**